### PR TITLE
Get AWS account id from sts user identity

### LINKED ISSUE #24: Hard-coded account_id might be an issue in multi-account environments
**Is this a BUG REPORT or FEATURE REQUEST?**:
Feature Request/Question

**What happened**:
Is a hard-coded account_id needed which might change in a multi-org setting having clusters across accounts. In this case the user has to change the parameter for every cluster and this might not be the case if there is a global pipeline deploying to multiple clusters.

**What you expected to happen**:
STS caller identity can get the account id based on the master-iam-role. With this there is no need to set the account-id.

**How to reproduce it (as minimally and precisely as possible)**:
Deploy application in a different account than the one mentioned in the configmap. Role creation fails.

**Anything else we need to know?**:

properties.go:

```
//LoadProperties function loads properties from various sources
func LoadProperties(ctx context.Context, kClient *k8s.Client, ns string, cmName string) (*Properties, error) {
	log := log.Logger(ctx, "config", "LoadProperties")
	log.WithValues("namespace", ns)
	log.Info("loading properties")
	stsClient := sts.New(session.New())
	callerIdentity, err := stsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
	if err != nil {
		log.Info("error getting caller idenity client. err: %v", err)
		return nil, err
	}
	return &Properties{
		AWSAccountId: *callerIdentity.Account,
	}, nil
}
```

**Environment**:
- iam-manager version
- Kubernetes version :
```
$ kubectl version -o yaml
```

**Other debugging information (if applicable)**:

```
- controller logs:
```
$ kubectl logs <iam-manager>
```

### PR DESCRIPTION
**Pull request by kshamajain99**
close #24

**Could you share the solution in high level?**

**Could you share the test results?**

> 2020-02-18T22:38:56.861Z	INFO	v1alpha1.Default	setting default version	{"name": "iamrole-sample"}
2020-02-18T22:38:56.863Z	INFO	v1alpha1.ValidateCreate	validating create request	{"name": "iamrole-sample"}
2020-02-18T22:38:56.863Z	INFO	v1alpha1.validateIAMPolicy	validating IAM policy	{"name": "iamrole-sample"}
2020-02-18T22:38:56.863Z	DEBUG	k8s.client.IamrolesCount	list api call
2020-02-18T22:38:56.866Z	INFO	k8s.client.IamrolesCount	Total number of roles	{"count": 0}
2020-02-18T22:38:56.873Z	INFO	controllers.iamrole_controller.Reconcile	Start of the request	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:56.873Z	INFO	controllers.iamrole_controller.Reconcile	New iamrole resource. Adding the finalizer	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6", "finalizer": "iamrole.finalizers.iammanager.keikoproj.io"}
2020-02-18T22:38:56.875Z	INFO	v1alpha1.Default	setting default version	{"name": "iamrole-sample"}
2020-02-18T22:38:56.877Z	INFO	v1alpha1.ValidateCreate	validate update	{"name": "iamrole-sample"}
2020-02-18T22:38:56.877Z	INFO	v1alpha1.validateIAMPolicy	validating IAM policy	{"name": "iamrole-sample"}
2020-02-18T22:38:56.877Z	DEBUG	k8s.client.IamrolesCount	list api call
2020-02-18T22:38:56.879Z	INFO	k8s.client.IamrolesCount	Total number of roles	{"count": 1}
2020-02-18T22:38:56.882Z	INFO	controllers.iamrole_controller.HandleReconcile	state of the custom resource 	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6", "state": ""}
2020-02-18T22:38:56.889Z	DEBUG	awsapi.iam.CreateRole	Initiating api call	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.021Z	DEBUG	awsapi.iam.CreateRole	Attaching Tag	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.021Z	DEBUG	awsapi.iam.TagRole	Initiating api call	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.131Z	DEBUG	awsapi.iam.TagRole	Successfully completed TagRole call	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.131Z	DEBUG	awsapi.iam.CreateRole	Attaching Permission Boundary	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.131Z	DEBUG	awsapi.iam.AddPermissionBoundary	Initiating api call	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.229Z	DEBUG	awsapi.iam.AddPermissionBoundary	Successfuly added permission boundary	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.229Z	DEBUG	awsapi.iam.CreateRole	Attaching Managed policies	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.230Z	DEBUG	awsapi.iam.AttachManagedRolePolicy	Initiating api call	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.341Z	DEBUG	awsapi.iam.AttachManagedRolePolicy	Successfully attached managed policies	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.341Z	DEBUG	awsapi.iam.CreateRole	Attaching Inline role policies	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.341Z	DEBUG	awsapi.iam.AttachInlineRolePolicy	Initiating api call	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.451Z	DEBUG	awsapi.iam.AttachInlineRolePolicy	Successfully completed attaching InlineRolePolicy	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.457Z	INFO	controllers.iamrole_controller.HandleReconcile	Successfully reconciled	{"request_id": "5628e240-f888-436f-a493-3156ddf2dbc6"}
2020-02-18T22:38:57.457Z	DEBUG	controller-runtime.controller	Successfully Reconciled	{"controller": "iamrole", "request": "this-is-my-new-test-namespace/iamrole-sample"}
2020-02-18T22:38:57.457Z	INFO	controllers.iamrole_controller.Reconcile	Start of the request	{"request_id": "1de90065-123e-4d37-bdd2-f78bcd8afff6"}
2020-02-18T22:38:57.457Z	INFO	controllers.iamrole_controller.HandleReconcile	state of the custom resource 	{"request_id": "1de90065-123e-4d37-bdd2-f78bcd8afff6", "state": "Ready"}
2020-02-18T22:38:57.457Z	DEBUG	awsapi.iam.GetRolePolicy	Initiating api call	{"request_id": "1de90065-123e-4d37-bdd2-f78bcd8afff6"}
2020-02-18T22:38:57.546Z	DEBUG	awsapi.iam.GetRolePolicy	Successfully able to get the policy	{"request_id": "1de90065-123e-4d37-bdd2-f78bcd8afff6"}
2020-02-18T22:38:57.546Z	INFO	controllers.iamrole_controller.comparePolicy	input policy and target policy are equal	{"request_id": "1de90065-123e-4d37-bdd2-f78bcd8afff6"}
2020-02-18T22:38:57.546Z	INFO	controllers.iamrole_controller.HandleReconcile	No change in the incoming policy compare to state of the world(external AWS IAM) policy	{"request_id": "1de90065-123e-4d37-bdd2-f78bcd8afff6"}
2020-02-18T22:38:57.546Z	INFO	controllers.iamrole_controller.HandleReconcile	Successfully reconciled	{"request_id": "1de90065-123e-4d37-bdd2-f78bcd8afff6"}
2020-02-18T22:38:57.546Z	DEBUG	controller-runtime.controller	Successfully Reconciled	{"controller": "iamrole", "request": "this-is-my-new-test-namespace/iamrole-sample"}

### CODE REVIEW

File: config/crd/bases/iammanager.keikoproj.io_iamroles.yaml [hunk 1]
```diff
@@ -47,370 +49,17 @@ spec:
             submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
           type: string
         metadata:
-          properties:
```
Comment (mnkg561) [inline config/crd/bases/iammanager.keikoproj.io_iamroles.yaml]: Do not commit anything from config folder until we figure out why it is removing all the content when you run make in your local machine.

File: pkg/awsapi/sts.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package awsapi
+
+//go:generate mockgen -destination=mocks/mock_stsiface.go -package=mock_awsapi github.com/aws/aws-sdk-go/service/sts/stsiface STSAPI
+//go:generate mockgen -destination=mocks/mock_sts.go -package=mock_awsapi github.com/keikoproj/iam-manager/pkg/awsapi STSIface
```
Comment (mnkg561) [inline pkg/awsapi/sts.go]: You probably don't need this line. First line is sufficient

File: pkg/awsapi/sts.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+type STSIface interface {
+	GetAccountID(ctx context.Context) (string, error)
+}
+
+type STS struct {
+	Client stsiface.STSAPI
+}
+
+func NewSTS() *STS {
+	sess, err := session.NewSession(&aws.Config{Region: aws.String("us-west-2")})
```
Comment (mnkg561) [inline pkg/awsapi/sts.go]: May be we should move region also into config map as everything else is now coming from cm and nothing is hard coded

Comment (kshamajain99) [reply pkg/awsapi/sts.go]: yup, true.

**Pull request merged**